cmd/operator/internal/controller/kubearchiveconfig_controller.go

```diff
@@ -202,51 +214,52 @@ func (r *KubeArchiveConfigReconciler) desiredRole(ctx context.Context, kaconfig
 	return role, nil
 }
 
-func (r *KubeArchiveConfigReconciler) reconcileRoleBinding(ctx context.Context, kaconfig *kubearchivev1alpha1.KubeArchiveConfig) (*rbacv1.RoleBinding, error) {
+func (r *KubeArchiveConfigReconciler) reconcileRoleBinding(ctx context.Context, kaconfig *kubearchivev1alpha1.KubeArchiveConfig, component string) (*rbacv1.RoleBinding, error) {
 	log := log.FromContext(ctx)
+	name := fmt.Sprintf("%s-%s", kaconfig.Name, component)
 
-	log.Info("in reconcileRoleBinding")
-	binding, err := r.desiredRoleBinding(kaconfig)
+	log.Info("in reconcileRoleBinding " + name)
+	binding, err := r.desiredRoleBinding(kaconfig, name)
 	if err != nil {
-		log.Error(err, "Unable to get desired RoleBinding")
+		log.Error(err, "Unable to get desired RoleBinding "+name)
 		return binding, err
 	}
 
-	err = r.Get(ctx, types.NamespacedName{Name: kaconfig.Name, Namespace: kaconfig.Namespace}, &rbacv1.RoleBinding{})
+	err = r.Get(ctx, types.NamespacedName{Name: name, Namespace: kaconfig.Namespace}, &rbacv1.RoleBinding{})
 	if err == nil {
 		err = r.Update(ctx, binding)
 		if err != nil {
-			log.Error(err, "Failed to update RoleBinding")
+			log.Error(err, "Failed to update RoleBinding "+name)
 			return binding, err
 		}
 	} else if errors.IsNotFound(err) {
 		err = r.Create(ctx, binding)
 		if err != nil {
-			log.Error(err, "Failed to create RoleBinding")
+			log.Error(err, "Failed to create RoleBinding "+name)
 			return binding, err
 		}
 	} else {
-		log.Error(err, "Failed to reconcile RoleBinding")
+		log.Error(err, "Failed to reconcile RoleBinding "+name)
 		return binding, err
 	}
 
 	return binding, nil
 }
 
-func (r *KubeArchiveConfigReconciler) desiredRoleBinding(kaconfig *kubearchivev1alpha1.KubeArchiveConfig) (*rbacv1.RoleBinding, error) {
+func (r *KubeArchiveConfigReconciler) desiredRoleBinding(kaconfig *kubearchivev1alpha1.KubeArchiveConfig, name string) (*rbacv1.RoleBinding, error) {
 	binding := &rbacv1.RoleBinding{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      kaconfig.Name,
+			Name:      name,
 			Namespace: kaconfig.Namespace,
 		},
 		RoleRef: rbacv1.RoleRef{
 			APIGroup: "rbac.authorization.k8s.io",
 			Kind:     "Role",
-			Name:     kaconfig.Name,
+			Name:     name,
 		},
 		Subjects: []rbacv1.Subject{{
 			Kind:      "ServiceAccount",
-			Name:      kaconfig.Name,
+			Name:      name,
 			Namespace: kaconfig.Namespace,
 		}},
```

Comment:
The ServiceAccount for the sink is the kubearchive namespace, not in the same namespace as the KubeArchiveConfig

Reply:
Ah, good catch!